src/Avalonia.Styling/Styling/Style.cs

```diff
@@ -111,17 +111,20 @@ public void Attach(IStyleable control, IStyleHost container)
                 {
                     var subs = GetSubscriptions(control);
 
-                    foreach (var animation in Animations)
+                    if (control is Animatable animatable)
                     {
-                        IObservable<bool> obsMatch = match.ObservableResult;
-
-                        if (match.ImmediateResult == true)
+                        foreach (var animation in Animations)
                         {
-                            obsMatch = Observable.Return(true);
-                        } 
+                            IObservable<bool> obsMatch = match.ObservableResult;
 
-                        var sub = animation.Apply((Animatable)control, obsMatch);
-                        subs.Add(sub);
+                            if (match.ImmediateResult == true)
+                            {
+                                obsMatch = Observable.Return(true);
+                            }
+
+                            var sub = animation.Apply(animatable, null, obsMatch);
```

Comment:
Shouldn't this use the control's `Clock` property?

Reply:
If we use the control's Clock property here then any value set on the `Clock` property after the style is applied will never be used. By determining the clock in `Animator.RunKeyFrames`, we allow the clock to be possibly changed.

Reply:
👍 makes sense